src/Device/Driver/LX_EOS/LXEosDevice.cpp

```diff
@@ -0,0 +1,199 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright The XCSoar Project
+
+#include "LXEosDevice.hpp"
+#include "Device/Driver.hpp"
+#include "Device/Driver/LX/Internal.hpp"
+#include "Device/Port/Port.hpp"
+#include "Device/Util/NMEAWriter.hpp"
+#include "NMEA/Checksum.hpp"
+#include "NMEA/Derived.hpp"
+#include "NMEA/DeviceInfo.hpp"
+#include "NMEA/Info.hpp"
+#include "NMEA/MoreData.hpp"
+#include "Units/Units.hpp"
+#include "util/ByteOrder.hxx"
+
+using std::string_view_literals::operator""sv;
+
+void
+LXEosDevice::LinkTimeout()
+{
+  const std::lock_guard lock{ settings_mutex };
+  vario_settings.uptodate = false;
+  altitude_offset.known = false;
+  altitude_offset.reliable = false;
+  vario_settings.device_polar.SetInvalid();
+  vario_settings.dry_mass_to_ref_mass_uptodate = false;
+}
+
+bool
+LXEosDevice::EnableNMEA(OperationEnvironment& env)
+{
+  /**
+   * Set up the NMEA sentences sent by the vario:
+   *
+   * - LXWP0 every second (most important data)
+   * - LXWP1 every 7 seconds (device info)
+   * - LXWP2 every 5 seconds (MC, Bugs, Ballast settings)
+   * - LXWP3 every 3 seconds (used for alt_offset (QNH) )
+   *
+   * It seems that LXWP3 does not get sent if it is to be sent at the same time
+   * as LXWP2, so choosing odd numbers to reduce overlaps
+   *
+   * LXWP2 is also sent automatically whenever vario settings are changed by
+   * pilot. Periodical sending is there only as a backup sync method.
+   */
+
+  // Set intervals of periodic messages
+  PortWriteNMEA(port, "PFLX0,LXWP0,1,LXWP1,7,LXWP2,5,LXWP3,3", env);
+  port.Flush();
+
+  return true;
+}
+
+bool
+LXEosDevice::PutMacCready(double mc, OperationEnvironment& env)
+{
+  const std::lock_guard lock{ settings_mutex };
+  vario_settings.mc = mc;
+  return SendNewSettings(env);
+}
+
+bool
+LXEosDevice::PutBallast([[maybe_unused]] double fraction,
```

Comment:
Beware, I think xcsoar is currently sending the wrong ratio to the external devices. Its falsley using dry mass instead of polar reference mass while talking to external devices. See my overload2 PR.

Reply:
Internally its using polar reference mass correctly.